src/app/pages/datasets/components/delete-dataset-dialog/delete-dataset-dialog.component.ts

```diff
@@ -56,53 +58,57 @@ export class DeleteDatasetDialogComponent implements OnInit {
 
   onDelete(): void {
     this.loader.open();
-    this.ws.call('pool.dataset.delete', [this.dataset.id, { recursive: true }])
-      .pipe(
-        catchError((error: WebsocketError) => {
-          if (error.reason.includes('Device busy')) {
-            return this.askToForceDelete();
-          }
-
-          return throwError(() => error);
-        }),
-        untilDestroyed(this),
-      )
-      .subscribe({
-        next: () => {
-          this.loader.close();
-          this.dialogRef.close(true);
-        },
-        error: (error) => {
-          this.dialog.errorReport(
-            this.translate.instant(
-              'Error deleting dataset {datasetName}.', { datasetName: this.dataset.name },
-            ),
-            error.reason,
-            error.stack,
-          );
-          this.loader.close();
-          this.dialogRef.close(true);
-        },
-      });
+
+    this.tryDeletingDataset().pipe(
+      catchError((error: WebsocketError) => {
+        if (error.reason.includes('Device busy')) {
+          return this.askToForceDelete();
+        }
+
+        return throwError(() => error);
+      }),
+      filter(Boolean),
+      switchMap(() => this.forceDeleteDataset()),
+      tap(() => {
+        this.loader.close();
+        this.dialogRef.close(true);
+      }),
+      catchError(this.handleDeleteError.bind(this)),
+      untilDestroyed(this),
+    ).subscribe();
+  }
+
+  tryDeletingDataset(): Observable<boolean> {
```

Comment:
I think `try` is a bit misleading here. I mean I get that in context of `onDelete` method we are trying, but there is nothing that would explain `try` in the method itself. I'd go with just `deleteDataset`.